ragna/deploy/_ui/components/metadata_filters_builder.py

```diff
@@ -0,0 +1,248 @@
+import panel as pn
+import param
+
+from ragna.core._metadata_filter import MetadataFilter, MetadataOperator
+
+EXCLUDED_OPERATORS = ["RAW", "AND", "OR"]
+ALLOWED_OPERATORS = [
+    op
+    for op in list(MetadataOperator.__members__.keys())
+    if op not in EXCLUDED_OPERATORS
+]
+
+
+class FilterRow(pn.viewable.Viewer):
+    key = param.Selector(objects=[""], default="")
+    operator = param.Selector(objects=[""] + ALLOWED_OPERATORS, default="")
+    value = param.Selector(objects=[""], default="")
+
+    def __init__(self, on_delete_callback, valid_key_value_pairs=None, **params):
+        super().__init__(**params)
+
+        key_select_disabled = True
+        if valid_key_value_pairs:
+            key_select_disabled = False
+            self.param.key.objects = [""] + list(valid_key_value_pairs.keys())
+            self.valid_key_value_pairs = valid_key_value_pairs
+
+        self.key_select = pn.widgets.Select.from_param(
+            self.param.key,
+            name="",
+            css_classes=["metadata-filter-row", "metadata-filter-key"],
+            disabled=key_select_disabled,
+        )
+
+        self.operator_select = pn.widgets.Select.from_param(
+            self.param.operator,
+            name="",
+            css_classes=["metadata-filter-row", "metadata-filter-operator"],
+            disabled=True,
+        )
+
+        self.value_select = pn.widgets.Select.from_param(
+            self.param.value,
+            name="",
+            css_classes=["metadata-filter-row", "metadata-filter-value"],
+            disabled=True,
+        )
+
+        self.delete_button = pn.widgets.ButtonIcon(
+            icon="trash",
+            width=25,
+            height=25,
+            css_classes=["metadata-filter-row", "metadata-filter-delete"],
+        )
+        self.delete_button.on_click(on_delete_callback)
+
+    @staticmethod
+    def set_css(widget, valid=True):
+        if valid:
+            widget.css_classes = [
+                c for c in widget.css_classes if c != "metadata-filter-row-error"
+            ]
+        else:
+            widget.css_classes = widget.css_classes + ["metadata-filter-row-error"]
+
+    @param.depends("key", watch=True)
+    def key_did_change(self):
```

Comment:
`Value` and `Operator` are only enabled, once a key has been selected. (So that we are able to provide a list of valid values to the user.)